rotkehlchen/api/v1/encoding.py

```diff
@@ -1453,6 +1476,59 @@ def _transform_ksm_address(
     return address
 
 
+def _transform_dot_address(
```

Comment:
It might be a good idea to combine this function and `_transform_ksm_address` using a new argument `chain` and two `if`s statements inside the function to avoid duplication

Reply:
I tried to integrate both functions but I am running into typing issues. Before, there was a `KusamaAddress` returned. Now, with combining the functions, I try to return a `SubstrateAddress` which already was defined as a Union of of `KusamaAddress` and `PolkadotAddress`

There is an error when trying to add a Polkadot Address now which points to the type error. Could you check? I spent quite some time but couldn't figure out a solution.

Reply:
@jonasW3F Okey Jonas. Will take a look today/tomorrow and see how this can be done :)

Reply:
This would do the job. I've noticed that you use double quotes instead of single. In our style we use always single `'` to declare strings. Also notice the `Literal` on the type of the chain argument so the linter knows that this function should only be called using this strings.

If your run the linter `make lint` you will see several things to correct regarding to imports mostly.

Finally there is a file conflict in the PR. Try to rebase and solve the conflict

```
def _transform_substrate_address(
        ethereum: EthereumManager,
        given_address: str,
        chain: Literal['Kusama', 'Polkadot'],
) -> SubstrateAddress:
    """Returns a DOT or KSM address (if exists) given an ENS domain. At this point any
    given address has been already validated either as an ENS name or as a
    valid Polkadot address (ss58 format).

    NB: ENS domains for Substrate chains (e.g. KSM, DOT) store the Substrate
    public key. It requires to encode it with a specific ss58 format for
    obtaining the specific chain address.

    Polkadot/Polkadot ENS domain accounts:
    https://guide.Polkadot.network/docs/en/mirror-ens

    ENS domain substrate public key encoding:
    https://github.com/ensdomains/address-encoder/blob/master/src/index.ts
    """
    if not given_address.endswith('.eth'):
        if chain == 'Polkadot':
            return PolkadotAddress(given_address)
        elif chain == 'Kusama':
            return KusamaAddress(given_address)

    try:
        if(chain == 'Polkadot'):
            resolved_address = ethereum.ens_lookup(
                given_address,
                blockchain=SupportedBlockchain.POLKADOT,
            )
        if(chain == 'Kusama'):
            resolved_address = ethereum.ens_lookup(
                given_address,
                blockchain=SupportedBlockchain.KUSAMA,
            )
    except (RemoteError, InputError) as e:
        raise ValidationError(
            f'Given ENS address {given_address} could not be resolved '
            f'for ' + chain + ' due to: {str(e)}',
            field_name='address',
        ) from None

    if resolved_address is None:
        raise ValidationError(
            f'Given ENS address {given_address} could not be resolved for ' + chain,
            field_name='address',
        ) from None

    try:
        if(chain == 'Polkadot'):
            address = get_polkadot_address_from_public_key(SubstratePublicKey(resolved_address))
            log.debug(f'Resolved polkadot ENS {given_address} to {address}')
            return PolkadotAddress(address)
        if(chain == 'Kusama'):
            address = get_kusama_address_from_public_key(SubstratePublicKey(resolved_address))
            log.debug(f'Resolved kusama ENS {given_address} to {address}')
            return KusamaAddress(address)
    except (TypeError, ValueError) as e:
        raise ValidationError(
            f'Given ENS address {given_address} does not contain a valid '
            f'Substrate public key: {resolved_address}. ' + chain + ' address cannot be obtained.',
            field_name='address',
        ) from e
```